python/src/aac/context/definition_parser.py

```diff
@@ -9,735 +9,841 @@
-
-            return inheritance_parents
-
-        def create_enum_class(enum_definition: Definition) -> Type:
-            """
-            Creates an enum class from a given enum definition.
-
-            Args:
-                enum_definition (Definition): An enum definition to convert to a class.
-
-            Returns:
-                The created class.
-            """
-            if not enum_definition.get_root_key() == "enum":
+                        # AaC only supports schema inheritance
+                        raise LanguageError(
+                            f"AaC extension is only supported for schema.  Unable to create parent class with AaC root: {parent_definition.get_root_key()}",
+                            self.get_location_str(parent_name, definition.lexemes),
+                        )
+
+        return inheritance_parents
+
+    def create_enum_class(self, enum_definition: Definition) -> Type:
+        """
+        Creates an enum class from a given enum definition.
+
+        Args:
+            enum_definition (Definition): An enum definition to convert to a class.
+
+        Returns:
+            The created class.
+        """
+        if not enum_definition.get_root_key() == "enum":
+            raise LanguageError(
+                f"Definition {enum_definition.name} is not an enum",
+                self.get_location_str(
+                    enum_definition.get_root_key(), enum_definition.lexemes
+                ),
+            )
+
+        fully_qualified_name = enum_definition.get_fully_qualified_name()
+        if (
+            fully_qualified_name
+            in self.context.context_instance.fully_qualified_name_to_class
+        ):
+            # we've already created the class, so nothing to do here
+            return self.context.context_instance.fully_qualified_name_to_class[
+                fully_qualified_name
+            ]
+
+        # Note:  trying to allow extension with Enums fails, so I just removed it
+        #        but we can revisit and try to find a solution in the future if needed
+
+        values = {}
+        if "values" in enum_definition.structure["enum"]:
+            for value in enum_definition.structure["enum"]["values"]:
+                values[value] = auto()
+        # create the enum class
+        instance_class = None
+        try:
+            instance_class = Enum(
+                enum_definition.get_python_class_name(),
+                values,
+                module=enum_definition.get_python_module_name(),
+            )
+        except LanguageError as e:
+            raise LanguageError(
+                f"Failed to create Enum instance_class for {enum_definition.name}: {e.message}",
+                self.get_location_str(enum_definition.name, enum_definition.lexemes),
+            )
+        self.context.context_instance.fully_qualified_name_to_class[
+            fully_qualified_name
+        ] = instance_class
+        return instance_class
```

Comment:
Because of the initial setting of `instance_class = None`, it does not like the return of the method definition being `Type` because this is now pushed into an `Any` type.